lib/IDE/CompletionLookup.cpp

```diff
@@ -873,19 +873,67 @@ void CompletionLookup::addVarDeclRef(const VarDecl *VD,
     Builder.addFlair(CodeCompletionFlairBit::ExpressionSpecific);
 }
 
+/// Return whether \p param has a non-desiable default value for code
+/// completion.
+///
+/// 'ClangImporter::Implementation::inferDefaultArgument()' automatically adds
+/// default values for some parameters;
+///   * NS_OPTIONS enum type with the name '...Options'.
+///   * NSDictionary and labeled 'options', 'attributes', or 'userInfo'.
+/// But code-completion doesn't want to make them defaulted when it is the
+/// first parameter, the parameter doesn't have an argument label, and the
+/// imported function base name ends with those words.
+///
+///   E.g. -(void)setOptions:(NSArray *) opts;
+///
+/// Note that '-(void)performWithOptions:(NSArray *) opts;' doen't match the
+/// condition because the base name of the function in Swift is 'peform'.
+bool isNonDesiableImportedDefaultArg(const ParamDecl *param) {
+  auto kind = param->getDefaultArgumentKind();
+  if (kind != DefaultArgumentKind::EmptyArray &&
+      kind != DefaultArgumentKind::EmptyDictionary)
+    return false;
+
+  if (!param->getArgumentName().empty())
+    return false;
+
+  auto *func = dyn_cast<FuncDecl>(param->getDeclContext());
+  if (!func->hasClangNode())
+    return false;
+  if (func->getParameters()->front() != param)
+    return false;
```

Comment:
Why do we only want to make them non-defaulted if they are the first argument?

Reply:
For
```objc
-(void)addAttributes:(NSDictionary *)attrs, options:(NSDictionary *)opts;
```
We want `attrs` _not_ have the default, but `opts` do have the default.

Reply:
Expanded the comment to explain the behavior.